File: GitHubPop/app/src/main/java/com/igormeira/githubpop/repository/RepositoryDataSourceFactory.java
package com.igormeira.githubpop.repository;

import androidx.lifecycle.MutableLiveData;
import androidx.paging.DataSource;

import com.igormeira.githubpop.model.Repository;

public class RepositoryDataSourceFactory extends DataSource.Factory<Long, Repository> {

    public MutableLiveData<RepositoryDataSource> userLiveDataSource = new MutableLiveData<>();

    @Override
    public DataSource<Long, Repository> create() {
        RepositoryDataSource dataSource = new RepositoryDataSource();
        userLiveDataSource.postValue(dataSource);
        return dataSource;
    }
}

File: app/src/main/java/dev/theuzfaleiro/trendingongithub/ui/feature/home/datasource/RepositoryDataSourceFactory.kt
package dev.theuzfaleiro.trendingongithub.ui.feature.home.datasource

import androidx.paging.DataSource
import dev.theuzfaleiro.trendingongithub.network.GitHubEndpoint
import dev.theuzfaleiro.trendingongithub.ui.feature.home.model.data.Repository

class RepositoryDataSourceFactory(private val gitHubEndpoint: GitHubEndpoint) :
    DataSource.Factory<Int, Repository>() {

    override fun create(): DataSource<Int, Repository> = RepositoryDataSource(gitHubEndpoint)
}